couchdb/couchdb.go

```diff
@@ -156,13 +170,17 @@ func makeRequest(method, path string, reqbody interface{}, resbody interface{})
 	return err
 }
 
+func fixErrorNoDatabaseIsWrongDoctype(err error) {
+	if IsNoDatabaseError(err) {
+		err.(*Error).Reason = "wrong_doctype"
+	}
+}
+
 // GetDoc fetch a document by its docType and ID, out is filled with
 // the document by json.Unmarshal-ing
 func GetDoc(dbprefix, doctype, id string, out Doc) error {
 	err := makeRequest("GET", docURL(dbprefix, doctype, id), nil, out)
-	if IsNoDatabaseError(err) {
-		err.(*Error).Reason = "wrong_doctype"
-	}
+	fixErrorNoDatabaseIsWrongDoctype(err)
```

Comment:
Maybe missing something but why not put this `fixErrorNoDatabaseIsWrongDoctype` at the end of `makeRequest` directly ?